exporter_test.go

```diff
@@ -640,6 +734,44 @@ type = "Apache-2.0"
 			})
 		})
 
+		when("previous slice image doesn't exist", func() {
+			var (
+				nonExistingOriginalImage *fakes.Image
+			)
+
+			it.Before(func() {
+				h.RecursiveCopy(t, filepath.Join("testdata", "exporter", "previous-slice-image-not-exist", "layers"), layersDir)
+				var err error
+				appDir, err = filepath.Abs(filepath.Join("testdata", "exporter", "previous-slice-image-not-exist", "layers", "app"))
+				h.AssertNil(t, err)
+
+				// TODO : this is an hacky way to create a non-existing image and should be improved in imgutil
+				nonExistingOriginalImage = fakes.NewImage("app/original-image", "", nil)
+				nonExistingOriginalImage.Delete()
+			})
+
+			it.After(func() {
+				nonExistingOriginalImage.Cleanup()
+			})
+
+			it("create a slice layer on the Run image", func() {
+				h.AssertNil(t, ioutil.WriteFile(filepath.Join(appDir, "static", "assets", "config.txt"), []byte("app-config"), os.ModePerm))
+				h.AssertNil(t, ioutil.WriteFile(filepath.Join(appDir, "static", "assets", "image.jpeg"), []byte("app-image"), os.ModePerm))
+
+				h.AssertNil(t, exporter.Export(layersDir, appDir, fakeAppImage, runImageRef, lifecycle.LayersMetadata{}, additionalNames, launcherConfig, stack))
+
+				sliceLayerPath, _ := fakeAppImage.FindLayerWithPath(filepath.Join(appDir, "static", "assets", "config.txt"))
+
+				assertTarFileExists(t, sliceLayerPath, filepath.Join(appDir, "static", "assets", "config.txt"), true)
+				assertTarFileExists(t, sliceLayerPath, filepath.Join(appDir, "static", "assets", "image.jpeg"), true)
+
+				appLayerPath, _ := fakeAppImage.FindLayerWithPath(filepath.Join(appDir, "static", "assets", "file.keep"))
+
+				assertTarFileExists(t, appLayerPath, filepath.Join(appDir, "static", "assets", "config.txt"), false)
```

Comment:
We should assert that the remaining (non-sliced) app files are in the appLayer (final slice)